src/dbnode/storage/index/block.go

```diff
@@ -123,6 +124,8 @@ type block struct {
 
 	compact blockCompact
 
+	mutableSealedBlock Block
```

Comment:
This sounds like a contradiction since we've traditionally defined sealed blocks as immutable. Maybe `addOnSealedBlock` or `subBlock` or something else?